pkg/agent/run.go

```diff
@@ -49,6 +49,9 @@ var StrictMode bool
 // Run starts the agent process
 func Run(cmd *cobra.Command, args []string) {
 	ctx := context.Background()
+	// map of the current data gatherers that were configured in the config
+	// this will dgs to live outside the fetching cycle
+	dataGatherers := map[string]datagatherer.DataGatherer{}
```

Comment:
I moved the data gatherers that are fetched outside the cycle, my worry was we didn't want to always re initilize the data gatherers like we did, at least the ones that have a cache.